flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/JobGraph.java

```diff
@@ -593,10 +596,37 @@ public void uploadUserArtifacts(InetSocketAddress blobServerAddress, Configurati
 						new DistributedCache.DistributedCacheEntry(
```

Comment:
Should we maybe distinguish between a `DistributedCacheEntry` which represents an accessible file and a `BlobServerStoredDistributedCacheEntry` which knows the `BlobKey` under which the file is stored. This would make the whole upload and download business a bit clearer.

Reply:
Then `isZipped` would only be part of the `BlobServerStoredDistributedCacheEntry`.